spec/unit/type/schedule_spec.rb

```diff
@@ -542,6 +552,10 @@ def min(method, count)
     end
   end
 
+  it "should raise an error if the weekday is an int higher than 6" do
+    expect { @schedule[:weekday] > 6 }.to raise_error(Puppet::Error)
```

Comment:
I think you need to actually set this to a value greater than 6 in order to see it error. The way this is written right now, `@schedule[:weekday]` hasn't been set to anything yet.